streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java

```diff
@@ -402,6 +405,34 @@ private void cleanGlobal(final String intermediateUserTopic) {
         Assert.assertEquals(0, exitCode);
     }
 
+    private void waitUntilUserTopicsGotDeleted() {
```

Comment:
Yeah - you are correct. Which further highlights that i think it is currently too difficult to follow.